nvflare/app_opt/pt/quantization/dequantizer.py

```diff
@@ -56,29 +56,28 @@ def dequantization(
         n_bytes_after = 0
         n_bytes_meta = 0
         n_quant_params = 0
-        for param_name in params:
+        for i, param_name in enumerate(params.keys()):
             source_data_type = source_datatype[param_name]
             if source_data_type == "bool":
                 continue
 
             if quantization_type != "adaquant":
                 # get the bits information
-                source_date_bits = int(re.findall(r"\d+", source_data_type)[0])
+                source_data_bits = int(re.findall(r"\d+", source_data_type)[0])
                 quantization_bits = int(re.findall(r"\d+", quantization_type)[0])
 
                 # only dequantize if the quantization type is lower than the source data type
-                if quantization_bits >= source_date_bits:
+                if quantization_bits >= source_data_bits:
                     self.log_info(
                         fl_ctx,
                         f"Skipping dequantization for {param_name}, quantization bit {quantization_type} >= source data bit {source_data_type}",
                     )
                     continue
             values = params[param_name]
             n_bytes_before += values.nbytes
-            param_quant_state = quant_state.get(param_name)
-            if param_quant_state is not None:
-                for item in param_quant_state.values():
-                    if isinstance(item, (np.ndarray, torch.Tensor)):
+            if quant_state[param_name]:
```

Comment:
@ZiyueXu77 what happen if this param_name is NOT in quant_state dict?
would that ever happen?

Reply:
I think it should not